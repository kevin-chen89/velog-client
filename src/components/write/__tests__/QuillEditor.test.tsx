import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import QuillEditor, { QuillEditorProps } from '../QuillEditor';

describe('QuillEditor', () => {
  const setup = (props: Partial<QuillEditorProps> = {}) => {
    const initialProps: QuillEditorProps = {
      onConvertEditorMode: () => undefined,
    };
    const utils = render(<QuillEditor {...initialProps} {...props} />);
    return {
      ...utils,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup({});
    expect(container).toMatchSnapshot();
  });
  it('calls onConvertEditorMode', async () => {
    const onConvertEditorMode = jest.fn();
    const { getByTestId, getByText } = setup({
      onConvertEditorMode,
    });
    const convertButton = getByTestId('mdconvert');
    fireEvent.click(convertButton);
    const confirmButton = getByText('확인');
    fireEvent.click(confirmButton);
    expect(onConvertEditorMode).toBeCalled();
  });
});
