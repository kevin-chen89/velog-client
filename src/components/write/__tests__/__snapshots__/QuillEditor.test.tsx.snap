// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuillEditor matches snapshot 1`] = `
<div>
  <div
    class="sc-fjdhpX jdbyiP"
    data-testid="quill"
  >
    <textarea
      class="sc-gzVnrw sc-jTzLTM dEPvcD"
      placeholder="제목을 입력하세요"
      style="height: 0px;"
      tabindex="1"
    />
    <div
      class="sc-jzJRlG bcnBQv"
    />
    <div
      class="sc-bdVaJa cuQacX"
      id="toolbar"
    >
      <div
        class="sc-bwzfXH bgRFUk"
      >
        <button
          class="ql-header sc-bxivhb ggrQMd"
          value="1"
        >
          <div
            class="sc-htpNat jrmEoq"
          >
            H
            <span>
              1
            </span>
          </div>
        </button>
        <button
          class="ql-header sc-bxivhb ggrQMd"
          value="2"
        >
          <div
            class="sc-htpNat jrmEoq"
          >
            H
            <span>
              2
            </span>
          </div>
        </button>
        <button
          class="ql-header sc-bxivhb ggrQMd"
          value="3"
        >
          <div
            class="sc-htpNat jrmEoq"
          >
            H
            <span>
              3
            </span>
          </div>
        </button>
        <button
          class="ql-header sc-bxivhb ggrQMd"
          value="4"
        >
          <div
            class="sc-htpNat jrmEoq"
          >
            H
            <span>
              4
            </span>
          </div>
        </button>
      </div>
      <div
        class="sc-ifAKCX eqJNNy"
      />
      <div
        class="sc-bwzfXH bgRFUk"
      >
        <button
          class="ql-bold sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M15.6 10.79c.97-.67 1.65-1.77 1.65-2.79 0-2.26-1.75-4-4-4H7v14h7.04c2.09 0 3.71-1.7 3.71-3.79 0-1.52-.86-2.82-2.15-3.42zM10 6.5h3c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-3v-3zm3.5 9H10v-3h3.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5z"
            />
          </svg>
        </button>
        <button
          class="ql-italic sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M10 4v3h2.21l-3.42 8H6v3h8v-3h-2.21l3.42-8H18V4z"
            />
          </svg>
        </button>
        <button
          class="ql-underline sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 17c3.31 0 6-2.69 6-6V3h-2.5v8c0 1.93-1.57 3.5-3.5 3.5S8.5 12.93 8.5 11V3H6v8c0 3.31 2.69 6 6 6zm-7 2v2h14v-2H5z"
            />
          </svg>
        </button>
        <button
          class="ql-strike sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M10 19h4v-3h-4v3zM5 4v3h5v3h4V7h5V4H5zM3 14h18v-2H3v2z"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-ifAKCX eqJNNy"
      />
      <div
        class="sc-bwzfXH bgRFUk"
      >
        <button
          class="ql-blockquote sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
            />
          </svg>
        </button>
        <button
          class="ql-link sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
        </button>
        <button
          class="ql-image sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
            />
          </svg>
        </button>
        <button
          class="ql-code-block sc-bxivhb ggrQMd"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-ifAKCX eqJNNy"
      />
      <div
        class="sc-bwzfXH bgRFUk"
      >
        <button
          class="sc-bxivhb ggrQMd"
          data-testid="mdconvert"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 640 512"
            width="1em"
          >
            <path
              d="M593.85 452.92H46.15C20.7 452.92 0 432.22 0 406.77V105.23c0-25.45 20.7-46.15 46.15-46.15h547.69c25.45 0 46.15 20.7 46.15 46.15v301.54c.01 25.45-20.69 46.15-46.14 46.15zm-440-92.3v-120l61.54 76.92 61.54-76.92v120h61.54V151.38h-61.54l-61.54 76.92-61.54-76.92H92.31v209.23h61.54zM566.15 256h-61.54V151.38h-61.54V256h-61.54l92.31 107.69L566.15 256z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="sc-cSHVUG izPnWC"
    >
      <div
        tabindex="2"
      />
    </div>
  </div>
</div>
`;
